packages/hub/src/lib/oauth-handle-redirect.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { describe, expect, it } from "vitest";
+import { TEST_COOKIE, TEST_HUB_URL } from "../test/consts";
+import { oauthLoginUrl } from "./oauth-login-url";
+import { oauthHandleRedirect } from "./oauth-handle-redirect";
+
+describe("oauthHandleRedirect", () => {
+	it("should work", async () => {
+		const localStorage = {
+			nonce: undefined,
+			codeVerifier: undefined,
+		};
+		const url = await oauthLoginUrl({
+			clientId: "dummy-app",
+			redirectUrl: "http://localhost:3000",
+			localStorage,
+			scopes: "openid profile email",
+			hubUrl: TEST_HUB_URL,
+		});
+		const resp = await fetch(url, {
+			method: "POST",
```

Comment:
how of curiosity, is there any reason to do a POST instead of a GET (or vice-versa) here?

Reply:
The POST is to grant the oauth app the authorization. GET works for subsequent calls, but not the first time ever